app/src/main/java/org/fossasia/openevent/app/common/di/module/BarcodeModule.java

```diff
@@ -0,0 +1,69 @@
+package org.fossasia.openevent.app.common.di.module;
+
+import android.content.Context;
+import android.util.SparseArray;
+
+import com.google.android.gms.vision.CameraSource;
+import com.google.android.gms.vision.Detector;
+import com.google.android.gms.vision.barcode.Barcode;
+import com.google.android.gms.vision.barcode.BarcodeDetector;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import io.reactivex.Notification;
+import io.reactivex.subjects.PublishSubject;
+
+@Module(includes = AndroidModule.class)
+public class BarcodeModule {
+
+    @Provides
+    @Singleton
+    @Named("barcodeEmitter")
+    PublishSubject<Notification<Barcode>> providesBarcodeEmitter() {
+        return PublishSubject.create();
+    }
+
+    @Provides
```

Comment:
Plus, no one will be able to take selfies then ( you'll lock the camera forever if you make it a singleton )